client/src/app/tabs/bpmn/BpmnEditor.js

```diff
@@ -146,7 +146,6 @@ export class BpmnEditor extends CachedComponent {
 
     [
       'import.done',
-      'saveXML.done',
```

Comment:
I believe handling changes on `saveXML.done` is required for proper dirty checking to work.

Could you verify / investigate this?

Reply:
Yes, I think this is the intended behavior, since `MultiSheetTab#handleChanged` is called after `saveXML` is done.

When the intended behavior was to trigger `handledChanged` after `saveXML` is done for handling the dirty checking, it's not working correctly atm. That's a general problem in the dirty state handling (https://github.com/camunda/camunda-modeler/issues/1006). 

To fix this problem the `lastXML` property has to be updated correctly. The change function called after `saveXML.done` was not handling this properly (as far as I understand it). But I think this has to be handled right after the [lastXML of the modeler ](https://github.com/camunda/camunda-modeler/blob/next/client/src/app/tabs/bpmn/BpmnEditor.js#L363) is set correctly.

Reply:
If we're reworking the dirty state handling anyway I'd ask you to create a test case that verifies we're not accidentally re-introducing this bug.

Reply:
just add some tests for assuring that `switchXML` will do no import with old xml and the getXML methods will do not import again.

The bug described in this issue is not testable 1:1 with the mocked editors and modelers we use in our test suite. We should discuss testing it via an integration test..

Reply:
If stuff cannot easily be tested without referring to an integration test, could you investigate how to an integration test could look like? How do people test electron apps, integration test style?

Maybe there is a simple way to improve our test coverage some day in the future.

Reply:
Sure, I'm going to open an issue for investigating for an integration test concept. This is definitely something important to do in case of the [overall refactoring](https://github.com/camunda/camunda-modeler/issues/866). 

What to do with this issue right now? Any work to do?

Reply:
Just opened an issue for this: https://github.com/camunda/camunda-modeler/issues/1025